src/main/cpp/kernels/SystemML.cu

```diff
@@ -35,12 +35,13 @@ nvcc -ptx -arch=sm_30 SystemML.cu
  */
 extern "C"
 __global__ void copy_u2l_dense(double* ret, int dim, int N) {
-	int ix = blockIdx.x * blockDim.x + threadIdx.x;
-	int iy = blockIdx.y * blockDim.y + threadIdx.y;
+	int tid = blockIdx.x * blockDim.x + threadIdx.x;
```

Comment:
Based on http://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capabilities and java's maximum size, this makes sense for dense kernels on device with [compute capability 3.0](https://github.com/nakul02/systemml/blob/b1ef11f86650ee79cb7c2a4b61201f4d25ff048e/src/main/cpp/kernels/SystemML.cu#L23).